protostar-rust/src/lib.rs

```diff
@@ -82,3 +91,174 @@ fn run_tests_in_file(
+            .exec()
+            .unwrap();
+
+        let result =
+            dependencies_for_package(&scarb_metadata, &PackageId::from(String::from("12345679")));
+        let err = result.unwrap_err();
+
+        assert!(format!("{}", err).contains("Failed to find metadata for package"));
+    }
+
+    #[test]
+    fn get_protostar_config_for_package() {
+        let temp = assert_fs::TempDir::new().unwrap();
+        temp.copy_from("example_package", &["**/*"]).unwrap();
+        let scarb_metadata = MetadataCommand::new()
+            .inherit_stderr()
+            .current_dir(temp.path())
+            .exec()
+            .unwrap();
+
+        let config =
+            protostar_config_for_package(&scarb_metadata, &scarb_metadata.workspace.members[0])
+                .unwrap();
+
+        assert_eq!(config, ProtostarTestConfig { exit_first: false });
+    }
+
+    #[test]
+    fn get_protostar_config_for_package_err_on_invalid_package() {
+        let temp = assert_fs::TempDir::new().unwrap();
+        temp.copy_from("example_package", &["**/*"]).unwrap();
+        let scarb_metadata = MetadataCommand::new()
+            .inherit_stderr()
+            .current_dir(temp.path())
+            .exec()
+            .unwrap();
+
+        let result = protostar_config_for_package(
+            &scarb_metadata,
+            &PackageId::from(String::from("12345679")),
+        );
+        let err = result.unwrap_err();
+
+        assert!(format!("{}", err).contains("Failed to find metadata for package"));
+    }
+
+    #[test]
+    fn get_protostar_config_for_package_err_on_missing_config() {
+        let temp = assert_fs::TempDir::new().unwrap();
+        temp.copy_from("example_package", &["**/*"]).unwrap();
+        let content = "[package]
+name = \"example_package\"
+version = \"0.1.0\"";
+        temp.child("Scarb.toml").write_str(content).unwrap();
+
+        let scarb_metadata = MetadataCommand::new()
+            .inherit_stderr()
+            .current_dir(temp.path())
+            .exec()
+            .unwrap();
+
+        let result =
+            protostar_config_for_package(&scarb_metadata, &scarb_metadata.workspace.members[0]);
+        let err = result.unwrap_err();
+
+        assert!(format!("{}", err).contains("Failed to find protostar config for package"));
+    }
+
+    #[test]
+    fn collecting_tests() {
+        let temp = assert_fs::TempDir::new().unwrap();
+        temp.copy_from("example_package", &["**/*"]).unwrap();
+        let tests_path = Utf8PathBuf::from_path_buf(temp.to_path_buf()).unwrap();
+
+        let tests = collect_tests_in_directory(&tests_path).unwrap();
+
+        assert!(tests.len() > 0);
+    }
+
+    #[test]
+    fn collecting_tests_err_on_invalid_dir() {
+        let tests_path = Utf8PathBuf::from("aaee");
+
+        let result = collect_tests_in_directory(&tests_path);
+        let err = result.unwrap_err();
+
+        assert!(format!("{}", err).contains("Failed to read directory at path"));
```

Comment:
I'll change it to use `to_string`. `assert_eq` cannot be used here because the actual error is like `Failed to read directory at path = path/to/file` so we cannot assert equality.